tlspuffin-claims/src/violation.rs

```diff
@@ -106,6 +107,14 @@ where
                         return Some("Mismatching ciphers");
                     }
 
+                    // TODO@MAX: Do you agree that this policy below will capture server authentication bypass attacks?
```

Comment:
I agree that if a trace succeeds (both puts are in a finished state) and those two do not match, then we have a MITM.

I'm not 100% sure whether an attack can be observed this was though. E.g. its unclear whether the `pkey` will always be initialized or not given the implementation which writes to it.
This would require a test case to validate this I think.

Maybe we could use a known wolfssl bug to verify this.